leku/src/main/java/com/schibstedspain/leku/search/LekuSearchTextWatcher.java

```diff
@@ -0,0 +1,36 @@
+package com.schibstedspain.leku.search;
+
+import android.text.Editable;
+import android.text.TextWatcher;
+
+abstract class LekuSearchTextWatcher implements TextWatcher {
+
+  private int minCharacters;
+
+  LekuSearchTextWatcher(int minCharacters) {
+    this.minCharacters = minCharacters;
+  }
+
+  @Override
+  public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+  }
+
+  @Override
+  public void onTextChanged(CharSequence charSequence, int start, int count, int after) {
+    if ("".equals(charSequence.toString())) {
+      onEmptyText();
+    } else if (charSequence.length() > minCharacters && after > count) {
+      onText(charSequence);
```

Comment:
What about calling this callback `onTextChanged`? It's strange to call it `onText`. On text what?